app/controllers/domains_controller.rb

```diff
@@ -0,0 +1,25 @@
+class DomainsController < ApplicationController
+  before_action :authenticate_user!
```

Comment:
O authenticate_user pode estar no application controller, ou em um base controller pra garantir que n vai ter repetição de codigo redundante toda vez que um novo controller for criado.